### ISSUE TITLE
Problem with macros: "Wrong number of args "

### ISSUE THREAD

**Issue by danielribeiro** [anmonteiro/lumo#30]
Lumo throws this warning when macros are defined. For example

```
Lumo 1.0.0-alpha3
ClojureScript 1.9.293
 Exit: Control+D or :cljs/quit or exit
cljs.user=> (defmacro one [] 1)
true
cljs.user=> (prn (one))
WARNING: Wrong number of args (0) passed to cljs.user/one at line 1 
1
nil
cljs.user=>
```

The warning is `WARNING: Wrong number of args (0) passed to cljs.user/one at line 1 `

**Comment by anmonteiro** [comment]
This is related to the way macros are handled in self-hosted ClojureScript. Mike Fikes has an excellent post explaining that macros need to be compiled in a separate compilation stage:
http://blog.fikesfarm.com/posts/2015-09-07-messing-with-macros-at-the-repl.html

TL;DR: this is how it works:
```clojure
Lumo 1.0.0
ClojureScript 1.9.293
 Exit: Control+D or :cljs/quit or exit
cljs.user=> (ns cljs.user$macros)
nil
cljs.user$macros=> (defmacro one [] 1)
true
cljs.user$macros=> (in-ns 'cljs.user)
cljs.user=> (cljs.user/one)
1
cljs.user=>
```

**Comment by anmonteiro** [comment]
I'm closing this as it's not a bug. If there's anything else I can help with let me know!

**Comment by danielribeiro** [comment]
ok
